### PR TITLE
Improve GL JS examples that use the Geocoding API

### PR DESCRIPTION
**Pull request by domlet**
Re: https://github.com/mapbox/mapbox-gl-js-docs/pull/388#issuecomment-678013630

* Improve several GL JS examples related to geocoding: 
  * Elaborate on descriptions.
  * Improve code comments, ie: break lines, use complete sentences.
  * Optimize example content, ie: change style to make a map thing easier to see.
* Include a link to [Geocoding API docs](https://docs.mapbox.com/api/search/geocoding/) on all examples that mention `mapbox-gl-geocoder`:

1. `forward-geocode-custom-data.md`
1. `mapbox-gl-geocoder-accept-coordinates.md`
1. `mapbox-gl-geocoder-custom-render.md`
1. `mapbox-gl-geocoder-limit-region.md`
1. `mapbox-gl-geocoder-no-map.md`
1. `mapbox-gl-geocoder-outside-the-map.md`
1. `mapbox-gl-geocoder-with-flyto.md`
1. `mapbox-gl-geocoder-with-language.md`
1. `mapbox-gl-geocoder.md`
1. `point-from-geocoder-result.md`

Update 3/22/2021: 
* All tests are passing! :)

### CODE REVIEW

File: docs/pages/example/forward-geocode-custom-data.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ prependJs:
 - "import iframe from './forward-geocode-custom-data.html?iframe';"
 ---
 
-Use the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) control to search for places using Mapbox Geocoding AP and supplement results from a local data source or function. This example matches Chicago Park names with text input.
+This example adds a geocoding control on top of a web map and supplements the places data available through the  [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/) with custom data from a local data source.
```
Comment (HeyStenson) [inline docs/pages/example/forward-geocode-custom-data.md:20]:
```suggestion
This example adds a geocoding control on top of a web map and supplements the place data available through the [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/) with custom data from a local data source. 
```

File: docs/pages/example/mapbox-gl-geocoder-custom-render.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ prependJs:
 - "import iframe from './mapbox-gl-geocoder-custom-render.html?iframe';"
 ---
 
-Use a custom HTML rendering function with the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) to customize the dropdown menu.
+This example uses a custom HTML rendering function to customize the dropdown menu for a geocoding control.
+
+It uses the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) control, which enables places search using the [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/).
```
Comment (HeyStenson) [inline docs/pages/example/mapbox-gl-geocoder-custom-render.md:22]:
```suggestion
This example uses a custom HTML rendering function to customize the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) dropdown menu.

The [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) plugin enables place search using the [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/).
```

File: docs/pages/example/mapbox-gl-geocoder-limit-region.md [hunk 3]
```diff
@@ -17,6 +17,10 @@ prependJs:
 - "import iframe from './mapbox-gl-geocoder-limit-region.html?iframe';"
 ---
 
-Use the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) control to search for places using Mapbox Geocoding API limiting results to a region.
+This example uses the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) control to enable users to search for places using [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/), limiting search results to a region.
+
+When it uses `map.addControl` to add the control to the map, it creates a `MapboxGeocoder` object and defines restrictions for search results by using the [optional parameters](https://docs.mapbox.com/api/search/geocoding/#forward-geocoding) `countries`, `bbox`, and `filter`.
+
+Search for a place like _Starbucks_ to see search results restricted to the specified region, the New South Wales in Australia.
```
Comment (HeyStenson) [inline docs/pages/example/mapbox-gl-geocoder-limit-region.md:24]:
```suggestion
Search for a place like _Starbucks_ to see search results restricted to the specified region, New South Wales in Australia.
```

File: docs/pages/example/mapbox-gl-geocoder-with-language.md [hunk 4]
```diff
@@ -17,6 +17,8 @@ prependJs:
 - "import iframe from './mapbox-gl-geocoder-with-language.html?iframe';"
 ---
 
-Localize the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) to set the UI language and improve result relevance in that language.
+This example localizes the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) control by setting the UI language and improving places search results relevant in that language.
+
+The `mapbox-gl-geocoder` plugin uses the [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/) to enable places search.
```
Comment (HeyStenson) [inline docs/pages/example/mapbox-gl-geocoder-with-language.md:22]:
```suggestion
The `mapbox-gl-geocoder` plugin uses the [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/) to enable place search.
```

File: docs/pages/example/mapbox-gl-geocoder.html [hunk 5]
```diff
@@ -1,12 +1,16 @@
+<!-- Load the `mapbox-gl-js` plugin. -->
```
Comment (HeyStenson) [inline docs/pages/example/mapbox-gl-geocoder.html:1]:
```suggestion
<!-- Load the `mapbox-gl-geocoder` plugin. -->
```

**Review by HeyStenson — APPROVED** [review]
Looks good! Left a few comments, let me know if you have any questions.

**Comment by HeyStenson** [comment]
@ryanhamley would you mind reviewing this? No hurry but sometime next week would be great. We'd also love it if we could get some 👀  from someone on @mapbox/search to evaluate the use of the geocoder plugin. Thanks all!

File: docs/pages/example/mapbox-gl-geocoder-with-language.md [hunk 6]
```diff
@@ -17,6 +17,8 @@ prependJs:
 - "import iframe from './mapbox-gl-geocoder-with-language.html?iframe';"
 ---
 
-Localize the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) to set the UI language and improve result relevance in that language.
+This example localizes the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) control by setting the UI language and improving places search results relevant in that language.
```
Comment (ryanhamley) [inline docs/pages/example/mapbox-gl-geocoder-with-language.md:20]: Is `improving places search results` what you wanted? I'm a bit confused what this is saying

**Review by ryanhamley — APPROVED** [review]
Looks good! I just had one nit about a comment

File: docs/pages/example/mapbox-gl-geocoder-with-language.md (hunk 6 above)
Comment (domlet) [reply docs/pages/example/mapbox-gl-geocoder-with-language.md]: Thanks @ryanhamley – here is my revision:

> This example localizes the [mapbox-gl-geocoder](https://github.com/mapbox/mapbox-gl-geocoder) control for the German language by using the `language` parameter, which sets the geocoder UI elements, controls the language of the text supplied in responses, and also affects result scoring.
> 
> The `mapbox-gl-geocoder` plugin uses the [Mapbox Geocoding API](https://docs.mapbox.com/api/search/geocoding/) to enable place search.
> 
> Search for a place like _olympic stadium_ to see results localized for German (`de-DE`).

Does that make better sense? Thanks!

**Pull request merged**